### ISSUE TITLE
ccsn_IIp, make_co_wd, make_o_ne_wd seem highly sensitive to the last few digits of a constant

### ISSUE THREAD

**Issue by adamjermyn** [MESAHub/mesa#319]
On commit `87b1d09` these tests break, whereas they were working a few commits earlier and nothing as far as I can tell changed that should have broken them. The commit in question just changed the numerical value of the constant `sqrt_2_div_3`, which should equal sqrt(2/3), by adding six more significant figures on the end.

At least ccsn_IIp starts working again when I revert that change. So I think these tests are likely just very sensitive to the precise numbers here, which is a sign that they may not be converged...

**Comment by adamjermyn** [comment]
Just to add: All three cases work again once that change is reverted, so this is definitely an extreme sensitivity.

**Comment by rjfarmer** [comment]
What inlist did ccsn_IIP break with the change?

**Comment by adamjermyn** [comment]
`inlist_shock_part1_header`

**Comment by adamjermyn** [comment]
Here's the log: https://logs.mesastar.org/87b1d09a749e77c3b200beada829d54b61784364/cannon_conroy_intel/ccsn_IIp/out.txt

**Comment by warrickball** [comment]
`magnetic_braking`!?  >:(

**Comment by rjfarmer** [comment]
I'm impressed you manged to make custom_colors break with mx_butterfly

**Comment by adamjermyn** [comment]
Yeah. I'm a little surprised at the cases that fell out of the hurricane.

It is comforting (???) that the three cases that broke with extra precision on `sqrt(2/3)` also broke on `mx_butterfly`. So at least the failure is broader than just that one constant.

**Comment by adamjermyn** [comment]
Currently on the `sqrt_2_3` branch only `make_o_ne_wd` is failing, but the other two haven't finished running yet (which I suspect means they'll fail too, given that the run on `sqrt_2_3` started around the same time as `mx_butterfly`).

**Comment by warrickball** [comment]
`mx-butterfly` also managed to break `timing`, which surprises me. I thought that was a fairly straightforward test too.

**Comment by adamjermyn** [comment]
Thanks @evbauer for running `mx-butterfly` repeatedly!

Looks like we probably just got unlucky on the first run. Of five runs, two only had one failure, two only had two failures, and one had 16 failures. Unfortunately the ones where 1/2 failed usually had distinct failures We had `make_co_wd`  on one, magnetic_braking on a second, `12M_pre_ms_to_core_collapse` and `ns_h` on a third, and `make_o_ne_wd` and diffusion_smoothness on the fourth. Those mostly (but not entirely) overlap with the 16 failure (the one that doesn't appear in the 16 is ns_h ). There's full overlap between the ones that fail when we increase precision on `sqrt_2_3` and the 16. Those three are `ccsn_IIp , `make_co_wd`, and `make_o_ne_wd`. So I think we want to focus on those three, but periodically do a repeated `mx-butterfly` run to check for sensitive cases.

**Comment by fxt44** [comment]
how does one invoke mx-butterfly?

**Comment by evbauer** [comment]
It's actually quite easy thanks to a clever design by Josiah that seeds the small randomness off the commit hash. So `mx-butterfly` basically ends up being a copy of `main`, except that the test suite runs with a random ~few percent change to mesh and time resolution. Just merge `main` into `mx-butterfly` to generate a new test. And possibly push more empty commits on top of that to test repeatedly.

**Comment by warrickball** [comment]
I merged `main` into `sqrt_2_3` again and, after fixing some unit test output, it's all [passing normally](https://testhub.mesastar.org/sqrt_2_3/commits/5aa0a74) and is almost finished [with `[ci optional]`](https://testhub.mesastar.org/sqrt_2_3/commits/2dcd516). Who knows how we "fixed" things but I'm going to merge `sqrt_2_3` into main if `12M_pre_ms_to_core_collapse` and `c13_pocket`, which were never problematic, pass.

**Comment by adamjermyn** [comment]
Awesome, thanks for checking this!
